### PR TITLE
feat: simple input case to request vulnerability data via purl

### PR DESCRIPTION
**Pull request by cpendery**
## 📝 Description
Enables users to pass in a `.csv` of CPEs, with an optional second item in each row of a pURL, to quickly match against the vulnerability database without requiring a complete sbom. This uses the same matches as Grype would use when processing an SBOM in order to have the results be consistent event though it may under report actual CPE matches

Closes https://github.com/anchore/grype/issues/757
Closes https://github.com/anchore/grype/issues/516

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ You can also explicitly specify the scheme to use:
     {{.appName}} sbom:path/to/syft.json                 read Syft JSON from path on disk
     {{.appName}} registry:yourrepo/yourimage:tag        pull image directly from a registry (no container runtime required)
     {{.appName}} att:attestation.json --key cosign.pub  explicitly use the input as an attestation
+    {{.appName}} csv:path/to/yourcsv                    read a CSV of CPE,PURL from a path on disk
```
Comment (spiffcs) [inline cmd/root.go:66]: Thanks for updating the documentation!

**Review by spiffcs — COMMENTED** [review]
Small nit on the var delcaration, but @wagoodman my only question on this PR is for you.

There is limited data that is available on the PURL side of this kind of input. For example, if a package does not have certain metadata associated with it then grype will not know which matcher to use to provide the most accurate result.

Are we ok introducing this kind of input where incomplete package information is considered a best-effort at providing a vulnerability report?

**Comment by wagoodman** [comment]
This also looks like it closes https://github.com/anchore/grype/issues/516

File: grype/pkg/csv_provider.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	csvReader := csv.NewReader(reader)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse csv: %w", err)
+	}
+
+	packages := []Package{}
+
+	for _, row := range records {
+		rawCpe := strings.TrimSpace(row[0])
```
Comment (wagoodman) [inline grype/pkg/csv_provider.go:56]: question: do we need CPEs? That is, could we accept a list of only pURLs?

Comment (cpendery) [reply grype/pkg/csv_provider.go:56]: We definitely don't need a cpe, as we can generate a cpe set using the information from a purl. Would you rather allow a cpe as optional or only allow purl?

**Comment by cpendery** [comment]
> ~This also looks like it closes https://github.com/anchore/grype/issues/516~
> 
> 
> 
> Actually, it looks like it almost covers this case. I'm curious what it would take to close the gap to cover "simple" input as well. The answer could be as simple as "pURLs are better and should be minimally used".

This could be changed to prioritize a purl. Since purl's include ecosystems, they provide everything you need to match and generate the cpes for.

Though distro would still be information needed for some matches though I think passing it in the through the cli flag should still be fine `--platform <string>`

File: cmd/root.go (hunk 1 above)
Comment (spiffcs) [reply cmd/root.go:66]: @wagoodman we might have talked about this offline, but what should this be renamed too if we don't want to fully support `cve` parsing as a grype input?

**Comment by cpendery** [comment]
> Functionality on this is amazing thank you again @cpendery.
> 
> 
> 
> The only comment I had is where we don't outright say we support `csv` since that might imply you can set arbitrary header/column information for an input. 
> 
> 
> 
> Requesting input from @wagoodman since I think he had some ideas on what this feature's input surface might ultimately look like

Sounds good to me. I agree `csv` is too broad. I'll wait on that review before changing it to be mandatory purl, optional cpe

**Review by spiffcs — APPROVED** [review]

**Pull request merged**